libyul/optimiser/Suite.cpp

```diff
@@ -87,6 +87,41 @@ using namespace std;
 using namespace solidity;
 using namespace solidity::yul;
 
+namespace
+{
+
+#ifdef OUTPUT_PERFORMANCE_METRICS
+void outputPerformanceMetrics(map<string, int64_t> const& _metrics)
+{
+	vector<pair<string, int64_t>> durations(_metrics.begin(), _metrics.end());
+	sort(
+		durations.begin(),
+		durations.end(),
+		[](pair<string, int64_t> const& _lhs, pair<string, int64_t> const& _rhs) -> bool
+		{
+			return _lhs.second < _rhs.second;
+		}
+	);
+	int64_t total = 0;
+	for (auto&& [step, count]: durations)
+		total += count;
+
+	cerr << "Performance metrics of optimizer steps" << endl;
+	cerr << "======================================" << endl;
+	for (auto&& [step, count]: durations)
+	{
+		double percentage = 100.0 * static_cast<double>(count) / static_cast<double>(total);
```

Comment:
This can divide by zero.

Reply:
I think that's a risk worth taking.

Reply:
If you divide by zero you'll create a hole in the Matrix :P